File: hotel-acuario-front/src/app/services/hotel.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Hotel } from '../models/hotel.model';
import { environment } from '../../environments/environment.development';

@Injectable({
  providedIn: 'root',
})
export class HotelService {
  private readonly base = `${environment.apiBase}/hotel`;

  constructor(private http: HttpClient) {}

  obtener(): Observable<Hotel> {
    return this.http.get<Hotel>(this.base);
  }

  actualizar(hotel: Hotel): Observable<Hotel> {
    return this.http.put<Hotel>(this.base, hotel);
  }
}

File: Hotel-backend/src/main/java/com/acuario/hotel/service/HotelService.java
package com.acuario.hotel.service;

import com.acuario.hotel.dto.HotelRequestDTO;
import com.acuario.hotel.dto.HotelResponseDTO;

public interface HotelService {
	HotelResponseDTO obtener();
	HotelResponseDTO actualizar(HotelRequestDTO dto);
}

